### DISCUSSION TITLE
Implementing an Oven

### DISCUSSION THREAD

**Discussion by daerSeebaer** [space-wizards/space-station-14#20410] [Feature Request]
So this discussion is intended to collect/coordinate the design and implementation of a new kitchen appliance: The oven.

## Why an Oven

- The kitchen design document (https://olddocs.spacestation14.com/en/design/kitchen , didn't find it in the new docs) mentions this
- The idea to have different ways of getting the foods crafted is nice
- Gameplay wise the feature "Heat something up until stuff happens" is implemented on some things. But a microwave currently does minimal amount of heating.

## Design
So my thoughts about an oven:
- It can be based largely on a Microwave:
  - Constructible Machine
  - Can store items in it
  - Has a UI

The UI should give a list of Items in the Oven, a possibility to eject one or all of them, and a possibility to set a temperature.
The UI may show the individual temperatures of items in the oven.

The Oven has an "internal temperature", that over time shifts towards the temperature defined through the UI. If we assume a good thermal regulation system, we can model this with a linear increase/decrease so the oven gets to target temps quickly.

All items in the oven will over time have their temperature and SolutionTemperatures approach the oven temperature. This can be modelled with T_Item_New = T_Item_Old + (T_Oven - T_Item) * k * FrameTime

The value of k should be chosen in a manner that gives good results from a balance perspective - not too high (or food will be cooked instantly), not too low (we don't want Microwave heating speeds). 0.1 - 0.2 looks like a good range for cooking temperatures. For a more realistic approach, a Thermal Energy System that combines the internal energy of the air in the oven, the power draw of the oven, and the heat capacity of the item in the oven might be chosen.

The oven should be constructible like any machine. For this we need an oven circuitboard.

Further ideas and input from people who might help implement this are welcome. (I will do it myself eventually but don't want to stop anyone who might get to this first.)

**Comment by deltanedas** [comment]
good related item would be a grill which functions like hot plate but:
- instead of heating reagents in a solution container it heats items directly
- supports like 4 items since grills can fit lots of food
- cooks meat better than bread which would burn i imagine, i have never grilled bread so i cant confirm or deny

with a ui to change target heat

advantage over oven would probably be faster cooking but its a grill so no timer or anything, if burning food is implemented it could burn if you leave it there too long

**Comment by CatMagic1** [comment]
gonna grill some bread this weekend

for SCIENCE
